src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -2338,14 +2338,20 @@ FixedwingPositionControl::task_main()
 					_att_sp.pitch_body = math::constrain(_att_sp.pitch_body, -_parameters.man_pitch_max_rad, _parameters.man_pitch_max_rad);
 				}
 
-				/* lazily publish the setpoint only once available */
-				if (_attitude_sp_pub != nullptr) {
-					/* publish the attitude setpoint */
-					orb_publish(_attitude_setpoint_id, _attitude_sp_pub, &_att_sp);
-
-				} else if (_attitude_setpoint_id) {
-					/* advertise and publish */
-					_attitude_sp_pub = orb_advertise(_attitude_setpoint_id, &_att_sp);
+				if (!(_control_mode.flag_control_offboard_enabled &&
+				      !(_control_mode.flag_control_position_enabled ||
```

Comment:
For altitude control mode we need fw_pos_ctrl to publish an attitude setpoint when position, velocity, acceleration flags are false. The roll comes from the stick input, but currently that's handled in fw_pos_ctrl.

What about other offboard setpoints with position or altitude?